ROSE2_geneMapper.py

```diff
@@ -460,14 +459,15 @@ def mapEnhancerToGeneTop(rankByBamFile, controlBamFile, genome, annotFile, enhan
 
     #get the chromLists from the various bams here
     cmd = 'samtools idxstats %s' % (rankByBamFile)
-    idxStats = subprocess.Popen(cmd,stdout=subprocess.PIPE,shell=True)
-    idxStats= idxStats.communicate()
-    bamChromList = [line.split('\t')[0] for line in idxStats[0].split('\n')[0:-2]]
+    idxStats = subprocess.Popen(cmd, stdout=subprocess.PIPE, shell=True,
+                                universal_newlines=True).communicate()[0]
```

Comment:
Generally it's not good practice to use shell=True, but probably for a small scientific pipeline this would be okay. Here is an example of not using shell=True: https://github.com/HPC-buildtest/buildtest-framework/blob/devel/buildtest/utils/command.py#L55